[KOGITO-85] - Add Keycloak integration for SSO

ricardozanini opened the pull request:
See:
https://issues.redhat.com/browse/KOGITO-85

In this PR we add the possibility to install Keycloak server with `KogitoInfra`, adding a dependency to Keycloak Operator.

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

ricardozanini commented:
This is a very simple use case.

By turning the flag to `true` in `Spec.InstallKeycloak` on `KogitoInfra` CR we have Keycloak deployed. Depends on [Keycloak Operator](https://github.com/keycloak/keycloak-operator):

```yaml
    - description: Represents a Keycloak server to provide SSO for Kogito Services
      displayName: Keycloak
      kind: Keycloak
      name: keycloaks.keycloak.org
      version: v1alpha1
```

There's a ton of file changes because Keycloak Operator dependency comes with a lot of vendor dependencies. (looking forward to upgrade to Go 1.13 and get rid of it)

sutaakar commented:
@ricardozanini I won't be able to make a review until next week. If it needs to be merged soon then you can skip my review.

radtriste commented:
I get some problems trying to install via OLM.
Could you please rebase and I will check again ? :)

radtriste commented:
Trying to install Operator via Yaml files (and keycloak operator present)
`E0116 09:36:42.668290       1 reflector.go:126] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1alpha1.Keycloak: keycloaks.keycloak.org is forbidden: User "system:serviceaccount:kogito-85:kogito-operator" cannot list resource "keycloaks" in API group "keycloak.org" in the namespace "kogito-85"`

ricardozanini commented:
It seems that OLM enforces needed permissions for the required CRDs. I’ll enforce them in the roles file as well. PR being rebased soon. (Commuting to office)

ricardozanini commented:
Rebased and fixed, thanks for your analysis, @radtriste. The image is available here:
https://quay.io/repository/ricardozanini/kogito-cloud-operator?tab=tags (0.8.0-rc1)

radtriste commented:
OK with yaml files, was able to launch add a Keycloak instance with kogitoinfra.

Still have the problem with OLM ... we can check that together tomorrow ;)

spolti commented on deploy/operator.yaml:
```diff
@@ -17,7 +17,7 @@ spec:
       containers:
         - name: kogito-cloud-operator
-          image: quay.io/kiegroup/kogito-cloud-operator:0.7.0-rc4
+          image: quay.io/kiecloud/kogito-cloud-operator:0.7.0-rc4
```
why change this?

ricardozanini replied:
I made a mistake
